### PR TITLE
[WIP] [SRVCOM-1911] Move to v1beta1 version of crds 

### PR DESCRIPTION
**Pull request by skonto**
- Fixes compatibility issues due to v1beta1 migration
- Adds a conversion webhook as part of the knative-operator deployment, updates csv accordingly
- Updates to the new crds, webhook strategy is replaced to `None` as OLM [auto-generates that part](https://gist.github.com/skonto/f1e2cbce94af1dcd5eaec0716a383218) via the csv webhook definitions and also olm creates a secret and a service on behalf of the user. In any case it is not recommended to have 
- Modifies the knative.dv/pkg webhoook certificate fetching logic to get the certificate from a tls secret instead of the Knative one. This is required in order to use the auto-generated OLM one which has its on ca and tls data.
- Tested for conversion and normal use.

TODO:
- Update to release-1.5 branch of knative operator where crds migration code has been moved to the [reconcilers](https://github.com/knative/operator/commit/8fa562c7edc14ab9005c277d3f87fdbf8f28fd52) in order to test it. Right now we are on 1.4 but since this is planned for 1.26 we should have that update available from a tag when we upgrade to 1.5. :pray: Note that OLM does not support a job in operator's olm manifests anyway and there is no post-install hook.
- test different scenarios
- clean up

### CODE REVIEW

File: hack/lib/serverless.bash [hunk 1]
```diff
@@ -212,7 +212,7 @@ function deploy_knativeeventing_cr {
 
   # Install Knative Eventing
   cat <<EOF > "$eventing_cr"
-apiVersion: operator.knative.dev/v1alpha1
+apiVersion: operator.knative.dev/v1beta1
```
Comment (pierDipi) [inline hack/lib/serverless.bash:215]: From a test perspective, I'd stick with v1alpha1 for at least one release

Comment (skonto) [reply hack/lib/serverless.bash:215]: Yes makes sense, I am bit eager to test the new ones for now.

File: olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -783,21 +815,46 @@ spec:
...
-            - v1alpha1
+            - v1beta1
           operations:
             - CREATE
             - UPDATE
           resources:
             - knativeservings
       sideEffects: None
       webhookPath: /mutate-knativeservings
+    - generateName: conversion.knativeservings.operator.knative.dev
```
Comment (matzew) [inline olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml:834]: This all was added manually,
and the "matching" conversion was removed from the actual CRD, and put here.

Mind adding some sort of comment for the why? Not sure that's really very transparent - at least I stumbled on this when reviewing the branch :smile_cat:

File: openshift-knative-operator/pkg/common/resources.go [hunk 3]
```diff
@@ -3,13 +3,13 @@ package common
...
-	operatorv1alpha1 "knative.dev/operator/pkg/apis/operator/v1alpha1"
+	"knative.dev/operator/pkg/apis/operator/base"
 )
 
 // EnsureContainerMemoryLimit makes sure the memory limit for the given container is set
 // to the specified amount if not otherwise configured already.
-func EnsureContainerMemoryLimit(s *operatorv1alpha1.CommonSpec, containerName string, memory resource.Quantity) {
-	for i, v := range s.Resources {
+func EnsureContainerMemoryLimit(s *base.CommonSpec, containerName string, memory resource.Quantity) {
+	for i, v := range s.DeprecatedResources {
```
Comment (matzew) [inline openshift-knative-operator/pkg/common/resources.go:12]: DeprecatedResources? what's new ? :sweat_smile:

File: olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (skonto) [reply olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml:834]: Yeah this comes from following the guide [here](https://olm.operatorframework.io/docs/advanced-tasks/adding-admission-and-conversion-webhooks/), [ocp docs](https://docs.openshift.com/container-platform/4.10/operators/operator_sdk/osdk-generating-csvs.html#olm-defining-csv-webhook_osdk-generating-csvs) and the discussion [here](https://coreos.slack.com/archives/C3VS0LV41/p1659474954314299). We don't auto-generate this part, not using the operator-sdk or anything.

**Comment by skonto** [comment]
```
1717f0f4edd7","knative.dev/key":"knativeservings.operator.knative.dev","duration":0.000027663,"error":"secret \"knative-operator-service-cert\" is missing \"ca-cert.pem\" key","stacktrace":"knative.dev/pkg/controller.(*Impl).handleErr\n\t/go/src/github.com/openshift-knative/serverless-operator/vendor/knative.dev/pkg/controller/controller.go:566\nknative.dev/pkg/controller.(*Impl).processNextWorkItem\n\t/go/src/github.com/openshift-knative/serverless-operator/vendor/knative.dev/pkg/controller/controller.go:543\nknative.dev/pkg/controller.
```
Need to fix secret key for olm secret which is `olmCAKey`.

File: templates/csv.yaml [hunk 4]
```diff
@@ -17,7 +17,7 @@ metadata:
           }
         },
         {
-          "apiVersion": "operator.knative.dev/v1alpha1",
+          "apiVersion": "operator.knative.dev/v1beta1",
```
Comment (matzew) [inline templates/csv.yaml:20]: perhaps we could move the `knative kafka` bits to v1beta1 as well - but in a separate PR?

Comment (skonto) [reply templates/csv.yaml:20]: Absolutely I thought about this too so we dont give the false impression that Kafka is left behind.

**Comment by skonto** [comment]
```
{  could not resolve inputs: could not determine inputs for step [input:openshift_knative-v1.4_knative-eventing-src]: could not resolve base image from openshift/knative-v1.4:knative-eventing-src: imagestreamtags.image.openshift.io "knative-v1.4:knative-eventing-src" not found}
```

**Comment by pierDipi** [comment]
I'm not sure, the image is there:
```
$ docker pull registry.ci.openshift.org/openshift/knative-v1.4:knative-eventing-src
knative-eventing-src: Pulling from openshift/knative-v1.4
2d473b07cdd5: Already exists 
64a95a7ce1d7: Downloading [==>                                                ]  15.12MB/353.3MB
6155768c41ef: Downloading [=============>                                     ]   36.2MB/131.3MB
05033c574b2c: Downloading [=======>                                           ]  14.01MB/97.21MB

**Comment by skonto** [comment]
Need to add the migration code:
```
postupgrade.go:35: "knativeeventings.operator.knative.dev" does not have a single stored version: {TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:knativeeventings.operator.knative.dev GenerateName: Namespace: SelfLink: UID:55433a08-cede-45f2-95b5-bf6392e47922 ResourceVersion:55210 Generation:3 CreationTimestamp:2022-08-09 13:08:02 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[app.kubernetes.io/part-of:knative-operator app.kubernetes.io/version:devel operator.knative.dev/release:devel operators.coreos.com/serverless-operator.openshift-serverless:] Annotations:map[operatorframework.io/installed-alongside-ca291400d1a39ca2:openshift-serverless/serverless-operator.v1.25.0 operatorframework.io/installed-alongside-d39eb700d73945a9:openshift-serverless/serverless-operator.v1.24.0] OwnerReferences:[] Finalizers:[] ClusterName: ManagedFields:[{Manager:Go-http-client Operation:Update APIVersion:apiextensions.k8s.io/v1 Time:2022-08-09 13:08:02 +0000 UTC FieldsType:FieldsV1 FieldsV1:{"f:status":{"f:acceptedNames":{"f:kind":{},"f:listKind":{},"f:plural":{},"f:singular":{}},"f:conditions":{"k:{\"type\":\"Established\"}":{".":{},"f:lastTransitionTime":{},"f:message":{},"f:reason":{},"f:status":{},"f:type":{}},"k:{\"type\":\"NamesAccepted\"}":{".":{},"f:lastTransitionTime":{},"f:message":{},"f:reason":{},"f:status":{},"f:type":{}}}}} Subresource:status} {Manager:catalog Operation:Update APIVersion:apiextensions.k8s.io/v1 Time:2022-08-09 13:18:36 +0000 UTC FieldsType:FieldsV1 F
```

**Comment by skonto** [comment]
closing this in favor #1695

**Pull request closed without merging**